### PR TITLE
ACT: implement 'goto type declaration' action

### LINKED ISSUE #2102: Implement *Go to type declaration* (Ctrl-Shift-B)
I'm used to it from the Scala plugin and I think it is quite useful. 

My guess is it shouldn't be too difficult to implement, as both of the sub-actions (find the type of symbol, go to declaration) are already implemented.

### PR DESCRIPTION
**Pull request by Undin**
Closes #2102

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/navigation/goto/RsTypeDeclarationProvider.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.rust.lang.core.types.type
+
+class RsTypeDeclarationProvider : TypeDeclarationProvider {
+
+    override fun getSymbolTypeDeclarations(element: PsiElement): Array<PsiElement>? {
+        val type = when (element) {
+            is RsFunction -> element.retType?.typeReference?.type
+            is RsFieldDecl -> element.typeReference?.type
+            is RsConstant -> element.typeReference?.type
+            is RsPatBinding -> element.type
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/navigation/goto/RsTypeDeclarationProvider.kt]: Hm, shouldn't this handle arbitrary RsExpressions?

Comment (Undin) [reply src/main/kotlin/org/rust/ide/navigation/goto/RsTypeDeclarationProvider.kt]: Looks like no
If I understand correctly corresponding action looks for only references, named elements and lookup elements(https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/codeInsight/navigation/actions/GotoTypeDeclarationAction.java#L112-L116)
I really don't know which elements are lookup elements, but it seems that arbitrary `RsExpr` don't belong to these groups

**Comment by Undin** [comment]
@matklad Do you have any other comments?

**Comment by Undin** [comment]
bors=matklad

**Comment by Undin** [comment]
bors r=matklad

**Pull request merged**